job-service-container/pom.xml

```diff
@@ -168,6 +173,34 @@
                 <!--<db.connection>jdbc:postgresql://${docker.host.address}:${postgres.db.port}</db.connection>-->
             </properties>
         </profile>
+        <profile>
+            <id>nightly-test</id>
+            <activation>
+                <property>
+                    <name>RE_BUILD_TYPE</name>
+                    <value>release</value>
+                </property>
+            </activation>
+        </profile>
```

Comment:
This profile was not meant to do anything. The would make pom execute all the tests including the performance IT. This was possible because of a default activation profile(continuous) being present which would exclude the performance IT.

But I liked the part where we can have the exclusion as part of plugin's configuration. So, now the release profile will just override the "excluded" tag's configuration to an empty exclude tag. You can refer the effective-poms generated by help plugin in both the cases below - 

> mvn help:effective-pom 

> i.e. default configuration. This should **exclude** the performance  IT

```xml
            <configuration>
              <systemPropertyVariables>
                <webservice.adminport>${webservice.adminport}</webservice.adminport>
                <docker.host.address>${docker.host.address}</docker.host.address>
                <rabbitmq.node.port>${rabbitmq.node.port}</rabbitmq.node.port>
                <rabbitmq.ctrl.port>${rabbitmq.ctrl.port}</rabbitmq.ctrl.port>
              </systemPropertyVariables>
              <environmentVariables>
                <CAF_APPNAME>test/jobservice</CAF_APPNAME>
                <CAF_CONFIG_PATH>/home/simran/codebase/mf/JobService/job-service/job-service-container/test-config</CAF_CONFIG_PATH>
                <CAF_TRACKING_PIPE>tracking-workflow-in</CAF_TRACKING_PIPE>
                <CAF_STATUS_CHECK_INTERVAL_SECONDS>5</CAF_STATUS_CHECK_INTERVAL_SECONDS>
                <CAF_INTEGRATIONTESTS_FS_PATH>\\${docker.host.address}@${webdav.apache.port}\webdav</CAF_INTEGRATIONTESTS_FS_PATH>
                <CAF_WEBSERVICE_URL>http://self:1234/job-service/v1</CAF_WEBSERVICE_URL>
                <JOB_SERVICE_DATABASE_APPNAME>Job Service</JOB_SERVICE_DATABASE_APPNAME>
                <JOB_SERVICE_DATABASE_URL>jdbc:postgresql://${docker.host.address}:${postgres.db.port}/jobservice</JOB_SERVICE_DATABASE_URL>
                <JOB_SERVICE_DATABASE_USERNAME>postgres</JOB_SERVICE_DATABASE_USERNAME>
                <JOB_SERVICE_DATABASE_PASSWORD>root</JOB_SERVICE_DATABASE_PASSWORD>
                <webserviceurl>http://${docker.host.address}:${webservice.adminport}/job-service/v1</webserviceurl>
                <healthcheckurl>http://${docker.host.address}:${webservice.healthcheckport}/healthcheck</healthcheckurl>
                <webserviceurlhttps>https://${docker.host.address}:${webservice.https.adminport}/job-service/v1</webserviceurlhttps>
              </environmentVariables>
              <excludes>
                <exclude>com.hpe.caf.services.job.api.performance.JobServicePerformanceIT</exclude>
              </excludes>
            </configuration>
```



> mvn help:effective-pom -P nightly-test

> This should **include** the performance IT

```xml
           <configuration>
              <excludes combine.self="override">
                <exclude />
              </excludes>
              <systemPropertyVariables>
                <webservice.adminport>${webservice.adminport}</webservice.adminport>
                <docker.host.address>${docker.host.address}</docker.host.address>
                <rabbitmq.node.port>${rabbitmq.node.port}</rabbitmq.node.port>
                <rabbitmq.ctrl.port>${rabbitmq.ctrl.port}</rabbitmq.ctrl.port>
              </systemPropertyVariables>
              <environmentVariables>
                <CAF_APPNAME>test/jobservice</CAF_APPNAME>
                <CAF_CONFIG_PATH>/home/simran/codebase/mf/JobService/job-service/job-service-container/test-config</CAF_CONFIG_PATH>
                <CAF_TRACKING_PIPE>tracking-workflow-in</CAF_TRACKING_PIPE>
                <CAF_STATUS_CHECK_INTERVAL_SECONDS>5</CAF_STATUS_CHECK_INTERVAL_SECONDS>
                <CAF_INTEGRATIONTESTS_FS_PATH>\\${docker.host.address}@${webdav.apache.port}\webdav</CAF_INTEGRATIONTESTS_FS_PATH>
                <CAF_WEBSERVICE_URL>http://self:1234/job-service/v1</CAF_WEBSERVICE_URL>
                <JOB_SERVICE_DATABASE_APPNAME>Job Service</JOB_SERVICE_DATABASE_APPNAME>
                <JOB_SERVICE_DATABASE_URL>jdbc:postgresql://${docker.host.address}:${postgres.db.port}/jobservice</JOB_SERVICE_DATABASE_URL>
                <JOB_SERVICE_DATABASE_USERNAME>postgres</JOB_SERVICE_DATABASE_USERNAME>
                <JOB_SERVICE_DATABASE_PASSWORD>root</JOB_SERVICE_DATABASE_PASSWORD>
                <webserviceurl>http://${docker.host.address}:${webservice.adminport}/job-service/v1</webserviceurl>
                <healthcheckurl>http://${docker.host.address}:${webservice.healthcheckport}/healthcheck</healthcheckurl>
                <webserviceurlhttps>https://${docker.host.address}:${webservice.https.adminport}/job-service/v1</webserviceurlhttps>
              </environmentVariables>
            </configuration>
```

Reply:
I don't think we should be completely excluding the test by default.  What I was expecting was the full 30,000 iteration check to be done for the release build and a quick 100 iteration check for the CI build.  The way I'd expect this to be implemented is to have a property in the pom file that is 100 by default (in the normal properties section) and 30,000 if the nightly-test profile is activated (i.e. by having a properties section in nightly-test profile).  The value of the property should be passed to the integration tests in much the same way that the `webservice.adminport` etc. are (i.e. as a system property or an environment variable).  Within the test you should then use [`System.getProperty(key)`](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#getProperty(java.lang.String)) or [`System.getenv(name)`](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#getenv(java.lang.String)) to retrieve the number of iterations to perform.

Reply:
P.S. I do like your exclusion/inclusion override as you have it now.  Even though I don't want to use it in this case I could imagine it being applicable in future cases across the product.

Reply:
Added a property task.table.deletion.count with default value 100 in properties section. This value gets overridden to 30000 if nightly-test profile is activated. Also moved the test back into JobServiceIT(deleted JobServicePerformanceIT) class since this test is no longer a performance only test and will be getting executed on CI builds with smaller count.